modules/programs/thunderbird.nix

```diff
@@ -0,0 +1,273 @@
+
+      profiles = mkOption {
+        type = with types;
+          attrsOf (submodule ({ config, name, ... }: {
+            options = {
+              name = mkOption {
+                type = types.str;
+                default = name;
+                readOnly = true;
+                description = "This profile's name.";
+              };
+
+              settings = mkOption {
+                type = with types; attrsOf (oneOf [ bool int str ]);
+                default = { };
+                example = literalExpression ''
+                  {
+                    "aaa" = "bbb";
+                  }
+                '';
+                description = ''
+                  Preferences to add to this profile's
+                  <filename>user.js</filename>.
+                '';
+              };
+
+              isDefault = mkOption {
+                type = types.bool;
+                default = false;
+                example = true;
+                description = ''
+                  Whether this is a default profile. There must be exactly one
+                  default profile.
+                '';
+              };
+            };
+          }));
+      };
+
+      settings = mkOption {
+        type = with types; attrsOf (oneOf [ bool int str ]);
+        default = { };
+        example = literalExpression ''
+          {
+            "general.useragent.override" = "";
+            "privacy.donottrackheader.enabled" = true;
+          }
+        '';
+        description = ''
+          Attribute set of Thunderbird preferences to be added to
+          all profiles.
+        '';
+      };
+
+      withExternalGnupg = mkOption {
+        type = types.bool;
+        default = false;
+        example = literalExpression "true";
+        description = "Allow using external GPG keys with GPGME.";
+      };
+    };
+
+    accounts.email.accounts = mkOption {
+      type = with types;
+        attrsOf (submodule {
+          options.thunderbird = {
+            enable =
+              mkEnableOption "the Thunderbird mail client for this account";
+
+            profiles = mkOption {
+              type = with types; listOf str;
+              default = [ ];
+              example = literalExpression ''
+                [ "profile1" "profile2" ]
+              '';
+              description = ''
+                List of Thunderbird profiles for which this account should be
+                enabled. If this list is empty (the default), this account will
+                be enabled for all declared profiles.
+              '';
+            };
+
+            settings = mkOption {
+              type = with types; functionTo (attrsOf (oneOf [ bool int str ]));
+              default = _: { };
+              defaultText = "_: { }";
+              example = literalExpression ''
+                id: {
+                  "mail.identity.id${"\${id}"}.protectSubject" = false;
+                  "mail.identity.id${"\${id}"}.autoEncryptDrafts" = false;
```

Comment:
I think you want `''` for the escaping here:
```suggestion
                  "mail.identity.id${"''${id}"}.protectSubject" = false;
                  "mail.identity.id${"''${id}"}.autoEncryptDrafts" = false;
```

Reply:
It's correct but confusing because there are two levels of interpolation, the inner one being inside `"`. Just remove one:
```suggestion
                  "mail.identity.id''${id}.protectSubject" = false;
                  "mail.identity.id''${id}.autoEncryptDrafts" = false;
```